lib/util/Ranges.v3

```diff
@@ -76,4 +76,40 @@ component Ranges {
 		}
 		return i;
 	}
+	// Performs binary search and returns the index.
```

Comment:
This code should follow the tab-per-indent level style of the surrounding Virgil code.